engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/ForeignEvalNode.java

```diff
@@ -104,42 +116,28 @@ private void parseJs() {
       CallTarget ct = EpbContext.get(this).getEnv().parsePublic(source);
       Object fn = rewrapNode.execute(ct.call(), inner, outer);
       foreign = insert(JsForeignNode.build(fn, argNames.length));
-    } catch (Throwable e) {
-      if (InteropLibrary.getUncached().isException(e)) {
-        parseError = rewrapExceptionNode.execute((AbstractTruffleException) e, inner, outer);
-      } else {
-        throw e;
-      }
     } finally {
       inner.leave(this, p);
     }
   }
 
   private void parsePy() {
-    try {
-      String args = Arrays.stream(argNames).collect(Collectors.joining(","));
-      String head =
-          "import polyglot\n"
-              + "@polyglot.export_value\n"
-              + "def polyglot_enso_python_eval("
-              + args
-              + "):\n";
-      String indentLines =
-          code.getForeignSource().lines().map(l -> "    " + l).collect(Collectors.joining("\n"));
-      Source source =
-          Source.newBuilder(code.getLanguage().getTruffleId(), head + indentLines, "").build();
-      EpbContext context = EpbContext.get(this);
-      CallTarget ct = context.getEnv().parsePublic(source);
-      ct.call();
-      Object fn = context.getEnv().importSymbol("polyglot_enso_python_eval");
-      foreign = insert(PyForeignNodeGen.create(fn));
-    } catch (Throwable e) {
-      if (InteropLibrary.getUncached().isException(e)) {
```

Comment:
The reason why this has not worked before is that once `parsePublic` fails with "Language ... not installed", it throws `IllegalStateException`, which is not an interop exception, therefore, would be rethrown later.